### PR TITLE
`explorer` URL handling

### LINKED ISSUE #303: `explorer` URL handling
#### What would you like to be added:
For all actions that provides transactions (e.g, Send, Add/Remove Liquidity), we need a way to manage the redirect or open the tx in the explorer. The URL will depend on the network. Whether it's Mainnet, testnet etc.

Example, when click on Send, it opens the Ocean Interface. There's a TX ID there that when you click, it will open the explorer URL.

Create a common method that will open the tx on the dedicated URL depending on the network.

Mainnet - https://mainnet.defichain.io/#/DFI/mainnet/tx/(tx id here)
Testnet - https://testnet.defichain.io/#/DFI/testnet/home/tx/(tx id here)
Playground - https://playground.defichain.com/v0.7/regtest/transactions/(tx id here)

Need to apply these in these components: 
1. OceanInterface 
2. TransactionDetail Page

### PR DESCRIPTION
**Pull request by kyleleow**
#### What kind of PR is this?:

/kind feature

#### What this PR does / why we need it:
- Create generic function retrieve corresponding URL to open explorer for transaction based on the current network

#### Which issue(s) does this PR fixes?:

Fixes #303

### CODE REVIEW

File: app/api/wallet/network.ts [hunk 1]
```diff
@@ -24,3 +24,19 @@ export function getBip32Option (envNetwork: EnvironmentNetwork): Bip32Options {
...
+export function getTxURLByNetwork (network: EnvironmentNetwork, txId: string): string {
+  switch (network) {
+    case EnvironmentNetwork.MainNet:
+      return 'https://mainnet.defichain.io/#/DFI/mainnet/tx/' + txId
+
+    case EnvironmentNetwork.TestNet:
+      return 'https://testnet.defichain.io/#/DFI/testnet/home/tx/' + txId
+
+    case EnvironmentNetwork.LocalPlayground:
+      return 'http://localhost:19553/v0.7/regtest/transactions/' + txId
```
Comment (thedoublejay) [inline app/api/wallet/network.ts:37]: this one will fail because it's not always v0.7

**Review by thedoublejay — CHANGES_REQUESTED** [review]

File: app/api/wallet/network.ts (hunk 1 above)
Comment (kyleleow) [reply app/api/wallet/network.ts:37]: any endpoint that we can call to retrieve the correct url path?

**Comment by kyleleow** [comment]
Will wait for @fuxingloh to update `https://defiscan.xyz/tx` endpoint to accept `network` and `txid` param before final integration

File: app/api/wallet/network.ts [hunk 2]
```diff
@@ -24,3 +24,25 @@ export function getBip32Option (envNetwork: EnvironmentNetwork): Bip32Options {
     wif: network.wifPrefix
   }
 }
+
+export function getTxURLByNetwork (network: EnvironmentNetwork, txId: string): string {
```
Comment (fuxingloh) [inline app/api/wallet/network.ts:28]: don't put it here, this is strictly for wallet implementation

File: app/api/wallet/network.ts [hunk 3]
```diff
@@ -24,3 +24,25 @@ export function getBip32Option (envNetwork: EnvironmentNetwork): Bip32Options {
...
+  const baseUrl = new URL('https://defiscan.xyz/tx/')
```
Comment (fuxingloh) [inline app/api/wallet/network.ts:29]: On second thought thinking we should structure it as such:

`https://defichain.xyz/transactions/${txid}?network=${network}`

File: app/api/wallet/network.ts (hunk 2 above)
Comment (fuxingloh) [reply app/api/wallet/network.ts:28]: should implement this at context level via:

```ts
const { getTransactionUrl } = useDeFiScan()
const url = getTransactionUrl(txid)
```

**Review by fuxingloh — CHANGES_REQUESTED** [review]

File: app/contexts/DeFiScanContext.tsx [hunk 4]
```diff
@@ -0,0 +1,52 @@
...
+
+  return (
+    <DeFiScanContext.Provider value={context}>
+      {props.children}
+    </DeFiScanContext.Provider>
+  )
+}
+
+function getTxURLByNetwork (network: EnvironmentNetwork, txid: string): string {
+  const baseUrl = new URL(`https://defiscan.xyz/transactions/${txid}`)
```
Comment (fuxingloh) [inline app/contexts/DeFiScanContext.tsx:34]:
```suggestion
  const baseUrl = new URL(`https://defiscan.live/transactions/${txid}`)
```

PS we are changing it again

**Review by fuxingloh — CHANGES_REQUESTED** [review]

File: app/api/wallet/network.ts (hunk 2 above)
Comment (kyleleow) [reply app/api/wallet/network.ts]: added [DeFiScan in context hook](https://github.com/DeFiCh/wallet/pull/404/files#diff-220a25fc8649f07414f4522acbf078ebccc8cbe3f062c4ae1ce65df06e7dc467)

let me know if i missed out anything

**Comment by fuxingloh** [comment]
Available when https://github.com/DeFiCh/scan/pull/113 get merged

File: app/api/wallet/network.test.ts [hunk 5]
```diff
@@ -1,5 +1,5 @@
 import { EnvironmentNetwork } from "../../environment";
-import { getBip32Option } from "./network";
```
Comment (fuxingloh) [inline app/api/wallet/network.test.ts:2]: revert change?

File: app/contexts/DeFiScanContext.tsx [hunk 6]
```diff
@@ -0,0 +1,52 @@
...
+import { EnvironmentNetwork } from '../environment'
+import { useNetworkContext } from './NetworkContext'
+
+interface DeFiScan {
+  getTransactionUrl: (txid: string) => string
+}
+
+const DeFiScanContext = createContext<DeFiScan>(undefined as any)
+
+export function useDeFiScan (): DeFiScan {
```
Comment (fuxingloh) [inline app/contexts/DeFiScanContext.tsx:11]:
```suggestion
export function useDeFiScanContext (): DeFiScan {
```

**Review by fuxingloh — CHANGES_REQUESTED** [review]

**Review by thedoublejay — APPROVED** [review]

**Review by fuxingloh — DISMISSED** [review]

**Review by thedoublejay — APPROVED** [review]

**Pull request merged**